sqlmesh/core/snapshot/evaluator.py

```diff
@@ -377,9 +334,14 @@ def audit(
                     execution_time=execution_time,
                     raise_exception=raise_exception,
                     deployability_index=deployability_index,
+                    wap_id=wap_id,
                     **kwargs,
                 )
             )
+
+        if wap_id is not None:
+            self._wap_publish_snapshot(snapshot, wap_id, deployability_index)
```

Comment:
@tobymao @eakmanrq how do you feel about publishing implicitly as part of the successful audit call? Or should we make it explicit and introduce the `publish` method to `SnapshotEvaluator` and call it externally?

Reply:
i think it's fine, because if you pass wap_id, you're opting into it

Reply:
Yeah seems fine.